client/packages/common/src/plugins/hooks/usePluginElements.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { ComponentPluginType } from '../types';
+import { RecordWithId } from '../../types/utility';
+import { usePluginProvider } from '../components';
+import { PluginLoader } from '../components';
+
+export function usePluginElements<T extends RecordWithId>({
+  type,
+  data,
+}: {
+  type: ComponentPluginType;
+  data?: T;
+}) {
+  const { getComponentPlugins } = usePluginProvider();
+  const plugins = getComponentPlugins(type);
+
+  return plugins.map(plugin =>
+    plugin.isLoaded ? (
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      React.createElement<{ data: any }>(plugin.Component, {
```

Comment:
another compromise in using `any` because I couldn't get the types to work as I'd like

Reply:
Thanks! that's very nice. I've taken the implementation and then tried to extend and get it working fully, but without success. It has picked up the InvoiceNode error though, thanks!

> Wondering, does plugin belong into common if it has references to invoices, etc?

No, it shouldn't be ideally. The problem is that I can't put it in `host` because that's just a container which calls the other packages. If we have packages importing host we can end up with a circular dependency.

> Also, RecordWithId doesn't seem to be needed (?)

No, it isn't - thanks